src/Core/System/SalesChannel/Entity/SalesChannelRepositoryProcessCriteriaEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\SalesChannel\Entity;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Event\GenericEvent;
+use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class SalesChannelRepositoryProcessCriteriaEvent extends Event implements ShopwareSalesChannelEvent, GenericEvent
```

Comment:
What about
```php
class SalesChannelEntitySearchedEvent extends EntitySearchedEvent
{
    public function __construct(Criteria $criteria, EntityDefinition $definition, SalesChannelContext $context)
    {
        parent::__construct($criteria, $definition, $context->getContext());
        ....
    }
        
     ....
 }
 ```
 That way it would match how it is handled in the "normal" repositories